File: dotnet/GitHub/Models/Commit.cs
using Microsoft.Kiota.Abstractions.Serialization;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System;
namespace GitHubTodoDemo.GitHub.Models {
    /// <summary>
    /// Commit
    /// </summary>
    public class Commit : IAdditionalDataHolder, IParsable {
        /// <summary>Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.</summary>
        public IDictionary<string, object> AdditionalData { get; set; }
        /// <summary>A GitHub user.</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public NullableSimpleUser? Author { get; set; }
#nullable restore
#else
        public NullableSimpleUser Author { get; set; }
#endif
        /// <summary>The comments_url property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? CommentsUrl { get; set; }
#nullable restore
#else
        public string CommentsUrl { get; set; }
#endif
        /// <summary>The commit property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Commit_commit? CommitProp { get; set; }
#nullable restore
#else
        public Commit_commit CommitProp { get; set; }
#endif
        /// <summary>A GitHub user.</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public NullableSimpleUser? Committer { get; set; }
#nullable restore
#else
        public NullableSimpleUser Committer { get; set; }
#endif
        /// <summary>The files property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<DiffEntry>? Files { get; set; }
#nullable restore
#else
        public List<DiffEntry> Files { get; set; }
#endif
        /// <summary>The html_url property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? HtmlUrl { get; set; }
#nullable restore
#else
        public string HtmlUrl { get; set; }
#endif
        /// <summary>The node_id property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? NodeId { get; set; }
#nullable restore
#else
        public string NodeId { get; set; }
#endif
        /// <summary>The parents property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public List<Commit_parents>? Parents { get; set; }
#nullable restore
#else
        public List<Commit_parents> Parents { get; set; }
#endif
        /// <summary>The sha property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Sha { get; set; }
#nullable restore
#else
        public string Sha { get; set; }
#endif
        /// <summary>The stats property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public Commit_stats? Stats { get; set; }
#nullable restore
#else
        public Commit_stats Stats { get; set; }
#endif
        /// <summary>The url property</summary>
#if NETSTANDARD2_1_OR_GREATER || NETCOREAPP3_1_OR_GREATER
#nullable enable
        public string? Url { get; set; }
#nullable restore
#else
        public string Url { get; set; }
#endif
        /// <summary>
        /// Instantiates a new Commit and sets the default values.
        /// </summary>
        public Commit() {
            AdditionalData = new Dictionary<string, object>();
        }
        /// <summary>
        /// Creates a new instance of the appropriate class based on discriminator value
        /// </summary>
        /// <param name="parseNode">The parse node to use to read the discriminator value and create the object</param>
        public static Commit CreateFromDiscriminatorValue(IParseNode parseNode) {
            _ = parseNode ?? throw new ArgumentNullException(nameof(parseNode));
            return new Commit();
        }
        /// <summary>
        /// The deserialization information for the current model
        /// </summary>
        public IDictionary<string, Action<IParseNode>> GetFieldDeserializers() {
            return new Dictionary<string, Action<IParseNode>> {
                {"author", n => { Author = n.GetObjectValue<NullableSimpleUser>(NullableSimpleUser.CreateFromDiscriminatorValue); } },
                {"comments_url", n => { CommentsUrl = n.GetStringValue(); } },
                {"commit", n => { CommitProp = n.GetObjectValue<Commit_commit>(Commit_commit.CreateFromDiscriminatorValue); } },
                {"committer", n => { Committer = n.GetObjectValue<NullableSimpleUser>(NullableSimpleUser.CreateFromDiscriminatorValue); } },
                {"files", n => { Files = n.GetCollectionOfObjectValues<DiffEntry>(DiffEntry.CreateFromDiscriminatorValue)?.ToList(); } },
                {"html_url", n => { HtmlUrl = n.GetStringValue(); } },
                {"node_id", n => { NodeId = n.GetStringValue(); } },
                {"parents", n => { Parents = n.GetCollectionOfObjectValues<Commit_parents>(Commit_parents.CreateFromDiscriminatorValue)?.ToList(); } },
                {"sha", n => { Sha = n.GetStringValue(); } },
                {"stats", n => { Stats = n.GetObjectValue<Commit_stats>(Commit_stats.CreateFromDiscriminatorValue); } },
                {"url", n => { Url = n.GetStringValue(); } },
            };
        }
        /// <summary>
        /// Serializes information the current object
        /// </summary>
        /// <param name="writer">Serialization writer to use to serialize this model</param>
        public void Serialize(ISerializationWriter writer) {
            _ = writer ?? throw new ArgumentNullException(nameof(writer));
            writer.WriteObjectValue<NullableSimpleUser>("author", Author);
            writer.WriteStringValue("comments_url", CommentsUrl);
            writer.WriteObjectValue<Commit_commit>("commit", CommitProp);
            writer.WriteObjectValue<NullableSimpleUser>("committer", Committer);
            writer.WriteCollectionOfObjectValues<DiffEntry>("files", Files);
            writer.WriteStringValue("html_url", HtmlUrl);
            writer.WriteStringValue("node_id", NodeId);
            writer.WriteCollectionOfObjectValues<Commit_parents>("parents", Parents);
            writer.WriteStringValue("sha", Sha);
            writer.WriteObjectValue<Commit_stats>("stats", Stats);
            writer.WriteStringValue("url", Url);
            writer.WriteAdditionalData(AdditionalData);
        }
    }
}

File: php/GitHub/Models/Commit.php
<?php

namespace Baywet\Githubtododemo\Github\Models;

use Microsoft\Kiota\Abstractions\Serialization\AdditionalDataHolder;
use Microsoft\Kiota\Abstractions\Serialization\Parsable;
use Microsoft\Kiota\Abstractions\Serialization\ParseNode;
use Microsoft\Kiota\Abstractions\Serialization\SerializationWriter;

/**
 * Commit
*/
class Commit implements AdditionalDataHolder, Parsable 
{
    /**
     * @var array<string, mixed>|null $additionalData Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
    */
    private ?array $additionalData = null;
    
    /**
     * @var NullableSimpleUser|null $author A GitHub user.
    */
    private ?NullableSimpleUser $author = null;
    
    /**
     * @var string|null $comments_url The comments_url property
    */
    private ?string $comments_url = null;
    
    /**
     * @var Commit_commit|null $commit The commit property
    */
    private ?Commit_commit $commit = null;
    
    /**
     * @var NullableSimpleUser|null $committer A GitHub user.
    */
    private ?NullableSimpleUser $committer = null;
    
    /**
     * @var array<DiffEntry>|null $files The files property
    */
    private ?array $files = null;
    
    /**
     * @var string|null $html_url The html_url property
    */
    private ?string $html_url = null;
    
    /**
     * @var string|null $node_id The node_id property
    */
    private ?string $node_id = null;
    
    /**
     * @var array<Commit_parents>|null $parents The parents property
    */
    private ?array $parents = null;
    
    /**
     * @var string|null $sha The sha property
    */
    private ?string $sha = null;
    
    /**
     * @var Commit_stats|null $stats The stats property
    */
    private ?Commit_stats $stats = null;
    
    /**
     * @var string|null $url The url property
    */
    private ?string $url = null;
    
    /**
     * Instantiates a new commit and sets the default values.
    */
    public function __construct() {
        $this->setAdditionalData([]);
    }

    /**
     * Creates a new instance of the appropriate class based on discriminator value
     * @param ParseNode $parseNode The parse node to use to read the discriminator value and create the object
     * @return Commit
    */
    public static function createFromDiscriminatorValue(ParseNode $parseNode): Commit {
        return new Commit();
    }

    /**
     * Gets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @return array<string, mixed>|null
    */
    public function getAdditionalData(): ?array {
        return $this->additionalData;
    }

    /**
     * Gets the author property value. A GitHub user.
     * @return NullableSimpleUser|null
    */
    public function getAuthor(): ?NullableSimpleUser {
        return $this->author;
    }

    /**
     * Gets the comments_url property value. The comments_url property
     * @return string|null
    */
    public function getCommentsUrl(): ?string {
        return $this->comments_url;
    }

    /**
     * Gets the commit property value. The commit property
     * @return Commit_commit|null
    */
    public function getCommit(): ?Commit_commit {
        return $this->commit;
    }

    /**
     * Gets the committer property value. A GitHub user.
     * @return NullableSimpleUser|null
    */
    public function getCommitter(): ?NullableSimpleUser {
        return $this->committer;
    }

    /**
     * The deserialization information for the current model
     * @return array<string, callable(ParseNode): void>
    */
    public function getFieldDeserializers(): array {
        $o = $this;
        return  [
            'author' => fn(ParseNode $n) => $o->setAuthor($n->getObjectValue([NullableSimpleUser::class, 'createFromDiscriminatorValue'])),
            'comments_url' => fn(ParseNode $n) => $o->setCommentsUrl($n->getStringValue()),
            'commit' => fn(ParseNode $n) => $o->setCommit($n->getObjectValue([Commit_commit::class, 'createFromDiscriminatorValue'])),
            'committer' => fn(ParseNode $n) => $o->setCommitter($n->getObjectValue([NullableSimpleUser::class, 'createFromDiscriminatorValue'])),
            'files' => fn(ParseNode $n) => $o->setFiles($n->getCollectionOfObjectValues([DiffEntry::class, 'createFromDiscriminatorValue'])),
            'html_url' => fn(ParseNode $n) => $o->setHtmlUrl($n->getStringValue()),
            'node_id' => fn(ParseNode $n) => $o->setNodeId($n->getStringValue()),
            'parents' => fn(ParseNode $n) => $o->setParents($n->getCollectionOfObjectValues([Commit_parents::class, 'createFromDiscriminatorValue'])),
            'sha' => fn(ParseNode $n) => $o->setSha($n->getStringValue()),
            'stats' => fn(ParseNode $n) => $o->setStats($n->getObjectValue([Commit_stats::class, 'createFromDiscriminatorValue'])),
            'url' => fn(ParseNode $n) => $o->setUrl($n->getStringValue()),
        ];
    }

    /**
     * Gets the files property value. The files property
     * @return array<DiffEntry>|null
    */
    public function getFiles(): ?array {
        return $this->files;
    }

    /**
     * Gets the html_url property value. The html_url property
     * @return string|null
    */
    public function getHtmlUrl(): ?string {
        return $this->html_url;
    }

    /**
     * Gets the node_id property value. The node_id property
     * @return string|null
    */
    public function getNodeId(): ?string {
        return $this->node_id;
    }

    /**
     * Gets the parents property value. The parents property
     * @return array<Commit_parents>|null
    */
    public function getParents(): ?array {
        return $this->parents;
    }

    /**
     * Gets the sha property value. The sha property
     * @return string|null
    */
    public function getSha(): ?string {
        return $this->sha;
    }

    /**
     * Gets the stats property value. The stats property
     * @return Commit_stats|null
    */
    public function getStats(): ?Commit_stats {
        return $this->stats;
    }

    /**
     * Gets the url property value. The url property
     * @return string|null
    */
    public function getUrl(): ?string {
        return $this->url;
    }

    /**
     * Serializes information the current object
     * @param SerializationWriter $writer Serialization writer to use to serialize this model
    */
    public function serialize(SerializationWriter $writer): void {
        $writer->writeObjectValue('author', $this->getAuthor());
        $writer->writeStringValue('comments_url', $this->getCommentsUrl());
        $writer->writeObjectValue('commit', $this->getCommit());
        $writer->writeObjectValue('committer', $this->getCommitter());
        $writer->writeCollectionOfObjectValues('files', $this->getFiles());
        $writer->writeStringValue('html_url', $this->getHtmlUrl());
        $writer->writeStringValue('node_id', $this->getNodeId());
        $writer->writeCollectionOfObjectValues('parents', $this->getParents());
        $writer->writeStringValue('sha', $this->getSha());
        $writer->writeObjectValue('stats', $this->getStats());
        $writer->writeStringValue('url', $this->getUrl());
        $writer->writeAdditionalData($this->getAdditionalData());
    }

    /**
     * Sets the AdditionalData property value. Stores additional data not described in the OpenAPI description found when deserializing. Can be used for serialization as well.
     * @param array<string,mixed> $value Value to set for the AdditionalData property.
    */
    public function setAdditionalData(?array $value): void {
        $this->additionalData = $value;
    }

    /**
     * Sets the author property value. A GitHub user.
     * @param NullableSimpleUser|null $value Value to set for the author property.
    */
    public function setAuthor(?NullableSimpleUser $value): void {
        $this->author = $value;
    }

    /**
     * Sets the comments_url property value. The comments_url property
     * @param string|null $value Value to set for the comments_url property.
    */
    public function setCommentsUrl(?string $value): void {
        $this->comments_url = $value;
    }

    /**
     * Sets the commit property value. The commit property
     * @param Commit_commit|null $value Value to set for the commit property.
    */
    public function setCommit(?Commit_commit $value): void {
        $this->commit = $value;
    }

    /**
     * Sets the committer property value. A GitHub user.
     * @param NullableSimpleUser|null $value Value to set for the committer property.
    */
    public function setCommitter(?NullableSimpleUser $value): void {
        $this->committer = $value;
    }

    /**
     * Sets the files property value. The files property
     * @param array<DiffEntry>|null $value Value to set for the files property.
    */
    public function setFiles(?array $value): void {
        $this->files = $value;
    }

    /**
     * Sets the html_url property value. The html_url property
     * @param string|null $value Value to set for the html_url property.
    */
    public function setHtmlUrl(?string $value): void {
        $this->html_url = $value;
    }

    /**
     * Sets the node_id property value. The node_id property
     * @param string|null $value Value to set for the node_id property.
    */
    public function setNodeId(?string $value): void {
        $this->node_id = $value;
    }

    /**
     * Sets the parents property value. The parents property
     * @param array<Commit_parents>|null $value Value to set for the parents property.
    */
    public function setParents(?array $value): void {
        $this->parents = $value;
    }

    /**
     * Sets the sha property value. The sha property
     * @param string|null $value Value to set for the sha property.
    */
    public function setSha(?string $value): void {
        $this->sha = $value;
    }

    /**
     * Sets the stats property value. The stats property
     * @param Commit_stats|null $value Value to set for the stats property.
    */
    public function setStats(?Commit_stats $value): void {
        $this->stats = $value;
    }

    /**
     * Sets the url property value. The url property
     * @param string|null $value Value to set for the url property.
    */
    public function setUrl(?string $value): void {
        $this->url = $value;
    }

}
